src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/TransmissionStatusTelemetryModule.cs

```diff
@@ -0,0 +1,149 @@
+    /// Transmission Status Telemetry Module 
+    /// </summary>
+    internal class TransmissionStatusTelemetryModule : ITelemetryModule, IDisposable
+    {
+        private bool isInitialized = false;
+        DiagnosticListener source = new DiagnosticListener(LoggingConstants.HostDiagnosticSourcePrefix + "ApplicationInsights");
+        string debugTracingEventName = LoggingConstants.HostDiagnosticSourceDebugEventNamePrefix + "TransmissionStatus";
+        JsonSerializerSettings options = new JsonSerializerSettings() { Error = (sender, error) => error.ErrorContext.Handled = true, NullValueHandling = NullValueHandling.Ignore };
+
+
+        /// <summary>
+        /// Initializes the telemetry module.
+        /// </summary>
+        /// <param name="configuration">Telemetry configuration to use for initialization.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            // Prevent the telemetry module from being initialized multiple times.
+            if (isInitialized)
+            {
+                return;
+            }
+            (configuration.TelemetryChannel as ServerTelemetryChannel).TransmissionStatusEvent += Handler;
+            isInitialized = true;
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        public void Dispose()
+        {
+            source.Dispose();
+        }
+
+        public void Handler(object sender, TransmissionStatusEventArgs args)
+        {
+            // Do not block the main thread
+            Task.Run(() =>
+            {
+                if (sender != null && args != null)
+                {
+                    // Always log if the response is non-success
+                    if (args.Response.StatusCode != 200)
+                    {
+                        var transmission = sender as Transmission;
+                        string id = null;
+                        if (transmission != null)
+                        {                        
+                            id = transmission.Id;
+                        }
+
+                        var log = new
+                        {
+                            statusCode = args.Response?.StatusCode,
+                            description = args.Response?.StatusDescription,
+                            id
+                        };
+                        source.Write("TransmissionStatus", JsonConvert.SerializeObject(log, options));
+                    }
+
+                    // Log everything if the debug trace feature flag is enabled
+                    if (source.IsEnabled(debugTracingEventName))
+                    {
+                        source.Write(debugTracingEventName, FormattedLog(sender, args));
+                    }
+                }
+            });
+        }
+        internal string FormattedLog(object sender, TransmissionStatusEventArgs args)
+        {
+            var transmission = sender as Transmission;
```

Comment:
nit: Pattern matching and return early to reduce one level of nesting.

````
if (!(sender is Transmission transmission))
{
    return "Unable to parse transmission status response";
}
````

Reply:
optimized code.